### PR TITLE
MM-21897: Fix failing actions in simpleController

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

- Create channels during the init phase.
- Populate teams and channels after login.
- Set the createAt time during creating a post, and fix the time search range
while searching for posts.
- Choose a channel randomly instead of the first channel.
- Added a viewUser action which populates the users in the store.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-21897

### CODE REVIEW

File: loadtest/user/userentity/actions.go [hunk 1]
```diff
@@ -32,11 +31,17 @@ func (ue *UserEntity) Login() error {
...
 	if resp.Error != nil {
 		return resp.Error
 	}
 
+	// We need to set user again because the user ID does not get set
+	// if a user is already signed up.
+	if err := ue.store.SetUser(loggedUser); err != nil {
+		return err
+	}
+
```
Comment (streamer45) [inline loadtest/user/userentity/actions.go]: I think this was the missing part to make the tests *resumable* without clearing the db every time.

File: loadtest/control/simplecontroller/actions.go [hunk 2]
```diff
@@ -586,6 +593,34 @@ func (c *SimpleController) reload(full bool) control.UserStatus {
...
+	member, err := c.user.Store().RandomChannelMember(channel.Id)
+	if err != nil {
+		return c.newErrorStatus(err)
+	}
+
+	// GetUsersByIds for that userid
+	_, err = c.user.GetUsersByIds([]string{member.UserId})
+	if err != nil {
+		return c.newErrorStatus(err)
+	}
```
Comment (streamer45) [inline loadtest/control/simplecontroller/actions.go]: Do we need to implement `GetUser` or does the webapp actually call `GetUsersByIds` with one id only?

File: loadtest/user/userentity/actions.go (hunk 1 above)
Comment (agnivade) [reply loadtest/user/userentity/actions.go]: Yep

File: loadtest/control/simplecontroller/actions.go (hunk 2 above)
Comment (agnivade) [reply loadtest/control/simplecontroller/actions.go]: Not sure. My grepping skills are failing me.

```
10:49:41-~/mattermost/mattermost-webapp$git grep "Client4\.getProfilesByIds"
10:49:45-~/mattermost/mattermost-webapp$git grep "Client4\.getUser"
components/suggestion/generic_user_provider.jsx:                    url={Client4.getUsersRoute() + '/' + item.id + '/image?_=' + (item.last_picture_update || 0)}
utils/utils.jsx:    return Client4.getUsersRoute() + '/' + userIdOrObject.id + '/image?_=' + (userIdOrObject.last_picture_update || 0);
utils/utils.jsx:    return Client4.getUsersRoute() + '/' + userId + '/image/default';
```

Comment (agnivade) [reply loadtest/control/simplecontroller/actions.go]: From my search, I think it uses `GetUsersInChannel` and `GetUsersInTeam`. I think `GetUsersInChannel` is the one we are looking for here. But it seems out of scope for this PR ? Perhaps another PR ?

Comment (streamer45) [reply loadtest/control/simplecontroller/actions.go]: > From my search, I think it uses `GetUsersInChannel` and `GetUsersInTeam`. I think `GetUsersInChannel` is the one we are looking for here. But it seems out of scope for this PR ? Perhaps another PR ?

I guess my doubt was about using:

https://godoc.org/github.com/mattermost/mattermost-server/model#Client4.GetUser

as opposed to:

https://github.com/mattermost/mattermost-server/blob/master/model/client4.go#L943

For fetching a single user like we do there.
But it's good for now, just curious.

Comment (agnivade) [reply loadtest/control/simplecontroller/actions.go]: I got that part. My point was that I think the webapp does not fetch a single user at all. It either fetches all users in a channel, or all users in a team. That's what I gleaned.

**Pull request merged**